### DISCUSSION TITLE
Screens: Access to query() returned data from layout()

### DISCUSSION THREAD

**Discussion by jfuentesa** [orchidsoftware/platform#2612] [Q&A]
Hello:

Is it possible to access the query returned data from the layout() method?

I'm using a instance attribute between the query() and layout() methods.

```
function query() {
    $this->queried = [
        ...content...
    ];
    
    return $this->queried;
}

public function layout(): iterable
{
    ... working with $this->queried....
}
```

**Comment by tabuna** [comment]
Hi @jfuentesa, I usually need some values in methods to check like "->canSee()" and branching.

That's what "Autocompleting Public Properties" is for. You just have to declare a public property with the same key name, and it will be automatically assigned. For example

```php
public $user;

public function query(User $user) : array
{
    return [
        'user' => $user,
    ];
}

public function layout(): iterable
{
    return [
        Layout::rows([
            Input::make('count')->readonly($this->user->anyBool)
        ])
    ];
}
```

For `Layout` classes you can use the `query` property to access data on the screen.
